INTRODUCTION.md

```diff
@@ -388,6 +392,244 @@ may be produced out of order, since a sub-sequent message in a sub-sequent
 ProduceRequest may already be in-flight (and accepted by the broker)
 by the time the retry for the failing message is sent.
 
+Using the Idempotent Producer prevents reordering even with `max.in.flight` > 1,
+see [Idempotent Producer](#idempotent-producer) below for more information.
+
+
+### Idempotent Producer
+
+librdkafka supports the idempotent producer which provides strict ordering and
+and exactly-once producer guarantees.
+The idempotent producer is enabled by setting the `enable.idempotence`
+configuration property to `true`, this will automatically adjust a number of
```

Comment:
the end of https://github.com/confluentinc/confluent-kafka-dotnet/pull/623#issuecomment-421098810 describes some flow on problems that this will cause.

Reply:
i was very good with the complaining, but failed to come up with something superior.